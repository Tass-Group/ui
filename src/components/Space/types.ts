export interface SpaceProps {
  children: React.ReactNode
  size?: number
  direction?: "horizontal" | "vertical"
}
