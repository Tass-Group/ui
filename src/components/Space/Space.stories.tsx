import Space from ".";
import Button from "../Button";
import { type SpaceProps } from "./types";

export default {
  title: "Layout/Space",
  component: Space,
  tags: ["autodocs"]

};

export const DefaultSpace = {
  render: (args: SpaceProps) => {
    return (
      <Space {...args}>
        <Button title="Button 1"/>
        <Button title="Button 2"/>
        <Button title="Button 3"/>
      </Space>
    );
  },
  args: {
    direction: "horizontal"
  }
};

export const VerticalSpace = {
  render: (args: SpaceProps) => {
    return (
      <Space {...args} >
        <Button title="Button 1"/>
        <Button title="Button 2"/>
        <Button title="Button 3"/>
      </Space>
    );
  },
  args: {
    direction: "vertical"
  }
};
