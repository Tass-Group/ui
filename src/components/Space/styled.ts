import styled from "styled-components";

export const SpaceContainer = styled.div<{
  size: number
  direction: "horizontal" | "vertical"
}>`
  display: flex;
  flex-direction: ${({ direction }) => direction === "horizontal" ? "row" : "column"};
  gap: ${({ size }) => `${size}px`};
`;
