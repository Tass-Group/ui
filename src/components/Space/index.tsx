import type React from "react";
import { SpaceContainer } from "./styled";
import { type SpaceProps } from "./types";

const Space: React.FC<SpaceProps> = ({ children, size = 8, direction = "horizontal" }) => {
  return <SpaceContainer size={size} direction={direction}>{children}</SpaceContainer>;
};

export default Space;
