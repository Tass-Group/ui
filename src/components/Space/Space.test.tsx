import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Space from ".";

describe("Space", () => {
  it("renders children correctly", () => {
    const { container } = render(
      <Space>
        <div>Child 1</div>
        <div>Child 2</div>
      </Space>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("applies default horizontal direction and gap", () => {
    const { container } = render(
      <Space>
        <div>Test</div>
      </Space>
    );
    const space = container.firstChild as HTMLElement;
    expect(space).toHaveStyle({
      display: "flex",
      flexDirection: "row",
      gap: "8px"
    });
  });

  it("applies vertical direction when specified", () => {
    const { container } = render(
      <Space direction="vertical">
        <div>Test</div>
      </Space>
    );
    const space = container.firstChild as HTMLElement;
    expect(space).toHaveStyle({
      flexDirection: "column"
    });
  });

  it("applies custom gap size", () => {
    const { container } = render(
      <Space size={16}>
        <div>Test</div>
      </Space>
    );
    const space = container.firstChild as HTMLElement;
    expect(space).toHaveStyle({
      gap: "16px"
    });
  });
});
