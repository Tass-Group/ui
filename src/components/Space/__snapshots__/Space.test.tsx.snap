// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space renders children correctly 1`] = `
<div
  class="sc-blHHSb txXxJ"
  direction="horizontal"
  size="8"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
</div>
`;
